### PR TITLE
Warn for `is` or `name` on dom-module

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/16)

### CODE REVIEW

File: src/html/dom-module-name-or-is.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
```
Comment (justinfagnani) [inline src/html/dom-module-name-or-is.ts]: 2017

File: src/html/dom-module-name-or-is.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  `);
+  constructor() {
+    super();
+  }
+
+  async checkDocument(document: ParsedHtmlDocument) {
+    const warnings: Warning[] = [];
+    const badModule = p.AND(
+        p.hasTagName('dom-module'), p.OR(p.hasAttr('is'), p.hasAttr('name')));
+    for (const domModule of dom5.nodeWalkAll(document.ast, badModule)) {
```
Comment (justinfagnani) [inline src/html/dom-module-name-or-is.ts]: nit: can you put the results of nodeWalkAll into a variable?

File: src/html/rule.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```
Comment (justinfagnani) [inline src/html/rule.ts]: 2017

File: src/html/rule.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import {Rule} from '../rule';
+
+
+/**
+ * An abstract rule that operates only over HTML files.
+ */
+export abstract class HtmlRule extends Rule {
+  async check(document: Document): Promise<Warning[]> {
+    const parsedDocument = document.parsedDocument;
+    if (!(parsedDocument instanceof ParsedHtmlDocument)) {
```
Comment (justinfagnani) [inline src/html/rule.ts:28]: Rules should probably specify why file types they work against, like parsers

File: src/html/rule.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      return [];
+    }
+    return this.checkDocument(parsedDocument, document);
+  }
+
+  abstract checkDocument(
```
Comment (justinfagnani) [inline src/html/rule.ts:37]: doc that this is the method that should be implemented, not `check()`

File: src/test/html/dom-module-name-or-is_test.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
```
Comment (justinfagnani) [inline src/test/html/dom-module-name-or-is_test.ts]: 2017

File: src/test/html/dom-module-name-or-is_test.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import {assert} from 'chai';
+import * as path from 'path';
+import {Analyzer} from 'polymer-analyzer';
+import {FSUrlLoader} from 'polymer-analyzer/lib/url-loader/fs-url-loader';
+import {WarningPrinter} from 'polymer-analyzer/lib/warning/warning-printer';
+
+import {DomModuleNameOrIs} from '../../html/dom-module-name-or-is';
+import {Linter} from '../../linter';
+
+const fixtures_dir = path.join(__dirname, '..', '..', '..', 'test');
```
Comment (justinfagnani) [inline src/test/html/dom-module-name-or-is_test.ts:24]: use path.resolve

File: src/test/html/dom-module-name-or-is_test.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  let analyzer: Analyzer;
+  let warningPrinter: WarningPrinter;
+  let linter: Linter;
+
+  setup(() => {
+    analyzer = new Analyzer({urlLoader: new FSUrlLoader(fixtures_dir)});
+    warningPrinter = new WarningPrinter(null as any, {analyzer: analyzer});
+    linter = new Linter([new DomModuleNameOrIs()], analyzer);
+  });
+  test('works in the trivial case', async() => {
```
Comment (justinfagnani) [inline src/test/html/dom-module-name-or-is_test.ts:37]: newline

File: src/util.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```
Comment (justinfagnani) [inline src/util.ts]: 2017

File: src/util.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export function stripWhitespace(str: string) {
```
Comment (justinfagnani) [inline src/util.ts:28]: docs

**Review by justinfagnani — APPROVED** [review]
LGTM with small nits

File: src/html/dom-module-name-or-is.ts (hunk 1 above)
Comment (rictic) [reply src/html/dom-module-name-or-is.ts]: Done

File: src/html/dom-module-name-or-is.ts (hunk 2 above)
Comment (rictic) [reply src/html/dom-module-name-or-is.ts]: Sure.

My argument in favor of this form is that it makes it clear that the result isn't used anywhere else. It's like `const` only more so.

File: src/html/rule.ts (hunk 3 above)
Comment (rictic) [reply src/html/rule.ts]: Done

File: src/html/rule.ts (hunk 4 above)
Comment (rictic) [reply src/html/rule.ts:28]: My thinking at the moment is that some rules might care about multiple kinds of document.

File: src/html/rule.ts (hunk 5 above)
Comment (rictic) [reply src/html/rule.ts:37]: Done

File: src/util.ts (hunk 9 above)
Comment (rictic) [reply src/util.ts]: Done

File: src/util.ts (hunk 10 above)
Comment (rictic) [reply src/util.ts:28]: Done

**Pull request merged**